File: chap1/Poste.py
# Ces programmes sont sous licence CeCILL-B V1.
type = input()
poids = int(input())
if type == "verte":
   if poids <= 20:
      print(0.57)
   elif poids <= 50:
      print(0.95)
   elif poids <= 100:
      print(1.40)
elif type == "prioritaire":
   if poids <= 20:
      print(0.60)
   elif poids <= 50:
      print(1.00)
   elif poids <= 100:
      print(1.45)
elif type == "ecopli":
   if poids <= 20:
      print(0.55)
   elif poids <= 50:
      print(0.78)
   elif poids <= 100:
      print(1.00)

File: chap1/Poste.java
// Ces programmes sont sous licence CeCILL-B V1.
class Poste {

  public static void main (String [] args) {
    String type; 
    int poids;
    type = Isn.readString ();
    poids = Isn.readInt (); 
    if (Isn.stringEqual(type,"verte")) {
      if (poids <= 20) {
        System.out.println(0.57);}
      else {
        if (poids <= 50) {
          System.out.println(0.95);}
        else {
          if (poids <= 100) {
            System.out.println(1.40);}}}}
    else {
      if (Isn.stringEqual(type,"prioritaire")) {
        if (poids <= 20) {
          System.out.println(0.60);}
        else {
          if (poids <= 50) {
            System.out.println(1.00);}
          else {
            if (poids <= 100) {
              System.out.println(1.45);}}}}
      else {
        if (Isn.stringEqual(type,"ecopli")) {
          if (poids <= 20) {
            System.out.println(0.55);}
          else {
            if (poids <= 50) {
              System.out.println(0.78);}
            else {
              if (poids <= 100) {
                System.out.println(1.00);}}}}}}}}
